primitives/src/pcs/univariate_kzg/mod.rs

```diff
@@ -910,4 +1152,77 @@ mod tests {
 
         Ok(())
     }
+
+    #[cfg(feature = "icicle")]
+    mod icicle {
+        use super::*;
+        use crate::{
+            icicle_deps::{curves::*, *},
+            pcs::univariate_kzg::icicle::GPUCommit,
+        };
+        use ark_ec::{short_weierstrass::Affine, CurveConfig};
+
+        fn test_gpu_e2e_template<E, C>() -> Result<(), PCSError>
+        where
+            C: IcicleCurve + MSM<C>,
+            C::ScalarField: ArkConvertible<ArkEquivalent = E::ScalarField>,
+            C::BaseField:
+                ArkConvertible<ArkEquivalent = <C::ArkSWConfig as CurveConfig>::BaseField>,
+            <C::ArkSWConfig as CurveConfig>::BaseField: PrimeField,
+            E: Pairing<G1Affine = Affine<<C as IcicleCurve>::ArkSWConfig>>,
+            UnivariateKzgPCS<E>: GPUCommit<E, C>,
+        {
+            let rng = &mut test_rng();
+            let stream = warmup_new_stream().unwrap();
+
+            let supported_degree = 2usize.pow(15);
+            let pp = UnivariateKzgPCS::<E>::gen_srs_for_testing(rng, supported_degree)?;
+            let (full_ck, _vk) = pp.trim(supported_degree)?;
+            let mut srs_on_gpu =
+                <UnivariateKzgPCS<E> as GPUCommit<E, C>>::load_prover_param_to_gpu(
+                    full_ck,
+                    supported_degree,
+                )?;
+
+            for _ in 0..10 {
+                let degree = usize::rand(rng) % 10000;
```

Comment:
I recommend running tests for large degrees, e.g. 2^20.

Reply:
done in [ca5b494](https://github.com/EspressoSystems/jellyfish/pull/498/commits/ca5b4942c0253dcda433abb2958f52e633cbae6e)